bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1255,6 +1259,30 @@ void ReplicaImp::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
     onReportAboutLateReplica(msgSenderId, msgSeqNum);
   }
 }
+/**
+ * Is sent from a read-only replica
+ */
+template <>
+void ReplicaImp::onMessage<AskForCheckpointMsg>(AskForCheckpointMsg *msg) {
```

Comment:
Don't you need to validate this message first?

Reply:
it's validated in ReplicaImp::messageHandler